src/database/standalone-sqlite.ts

```diff
@@ -2924,6 +2968,17 @@ export class StandaloneSqliteDatabase
     parentId: string;
     dataOffset: number;
   }): Promise<void> {
+    const key = `${hash}:${parentId}`;
+
+    if (this.saveNestedDataHashCache.get(key)) {
```

Comment:
PiKNiK testing matches your test results, @karlprieb. We see some bursts of duplicates, but overall there aren't many here. Let's remove the deduplication here and save some memory.